external-builds/pytorch/build_prod_wheels.py

```diff
@@ -601,10 +602,49 @@ def do_build_pytorch(
     # Compute version.
     pytorch_build_version = (pytorch_dir / "version.txt").read_text().strip()
     pytorch_build_version += args.version_suffix
+    pytorch_build_version_parsed = parse(pytorch_build_version)
     print(f"  Default PYTORCH_BUILD_VERSION: {pytorch_build_version}")
-    print(
-        f"  Flash attention enabled: {args.enable_pytorch_flash_attention_windows or not is_windows}"
-    )
+
+    ## Disable FBGEMM_GENAI and flash_attention only for Linux on 2.10 and higher Pytorch version
+    ## https://github.com/ROCm/TheRock/issues/1619
+    if not is_windows:
+        # Enabling/Disabling FBGEMM_GENAI based on Pytorch version in Linux
+        if pytorch_build_version_parsed.release < (2, 10):
+            env["USE_FBGEMM_GENAI"] = "ON"
+            print(
+                f"FBGEMM_GENAI enabled (PyTorch < 2.10, Linux): {env['USE_FBGEMM_GENAI'] == 'ON'}"
+            )
+        else:
+            env["USE_FBGEMM_GENAI"] = "OFF"
+            print(
+                f"FBGEMM_GENAI enabled (PyTorch >= 2.10, Linux): {env['USE_FBGEMM_GENAI'] == 'ON'}"
+            )
+
+        # Enabling/Disabling Flash attention based on Pytorch version in Linux
+        if pytorch_build_version_parsed.release < (2, 10):
+            use_flash_attention = (
+                "1" if args.enable_pytorch_flash_attention_linux else "0"
+            )
+            env.update(
+                {
+                    "USE_FLASH_ATTENTION": use_flash_attention,
+                    "USE_MEM_EFF_ATTENTION": use_flash_attention,
+                }
+            )
+            print(
+                f"Flash Attention enabled (PyTorch < 2.10, Linux): {env['USE_FLASH_ATTENTION'] == '1'}"
+            )
+        else:
+            env.update(
+                {
+                    "USE_FLASH_ATTENTION": "0",
+                    "USE_MEM_EFF_ATTENTION": "0",
+                }
+            )
+            print(
+                f"Flash Attention enabled (PyTorch >= 2.10, Linux): {env['USE_FLASH_ATTENTION'] == '1'}"
+            )
```

Comment:
I don't understand the use of `--enable-pytorch-flash-attention-linux` here. We want it enabled by default, except where it is known broken. This code currently
* On torch < 2.10 enables it by default but allows disabling it
* On torch >= 2.10 disables it always

Why even add an argument in that case? Shouldn't the conditions be flipped like so?

* On torch < 2.10 enable it always
* On torch >= 2.10 disable it by default but allow enabling it

Or

* On torch < 2.10 enable it by default but allow disabling it
* On torch >= 2.10 disable it by default but allow enabling it

The argument would then let developers try to enable it without needing to edit the build script.